PAINTROID-542 Fix Fill-Tool creating checkered pattern on erased parts

khalid-nasralla opened the pull request:
[PAINTROID-542](https://jira.catrob.at/browse/PAINTROID-542)
Fixed bug where filling an empty spot previously created with the Eraser tool created a checkered pattern/filled only one cell of the checkered pattern. Now it fills the the whole shape correctly.
Also added a Test to verify if it works correctly.

The ColorPicker is not clickable anymore when using the EraseTool as it is supposed to erase and have the checkeredPattern selected, otherwise it would be the same functionality as the brush tool, which wouldnt make much sense. This also fixed the issue with the visibility of the strokes and also makes more sense for the EraserTool.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

Lenkomotive reviewed (changes requested):
The stroke of the eraser is not visible when erasing. Please take a look at that again. Cheers

Lenkomotive commented on Paintroid/src/main/java/org/catrobat/paintroid/controller/DefaultToolController.kt:
```diff
@@ -119,6 +120,9 @@ class DefaultToolController(
         val currentTool = toolReference.tool
         val currentToolType = currentTool?.toolType
+        if (currentToolType == ToolType.ERASER) {
+            (currentTool as EraserTool).setSavedColor()
```
I like the idea! Please also add a testcase for your new implementation.  Cheers

khalid-nasralla replied:
> The stroke of the eraser is not visible when erasing. Please take a look at that again. Cheers

Strokes with the erase tool are now visible again during the strokes.

khalid-nasralla replied:
I also added some integration tests now

Lenkomotive reviewed (changes requested):
pls see comment

Lenkomotive commented on Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt:
```diff
@@ -207,12 +208,20 @@ class MainActivityNavigator(
+                val eraserColor: Int
+                if (toolReference.tool?.toolType == ToolType.ERASER) {
+                    eraserColor = Color.TRANSPARENT
+                    it.drawPaint.color = eraserColor
```
since eraserColor is used only once you could also just assign the value directly and save a variable there

Lenkomotive commented on Paintroid/src/main/java/org/catrobat/paintroid/ui/MainActivityNavigator.kt:
```diff
@@ -207,12 +208,20 @@ class MainActivityNavigator(
                     mainActivity.model.isOpenedFromFormulaEditorInCatroid,
                     mainActivity.model.colorHistory
                 )
+                if (toolReference.tool?.toolType != ToolType.ERASER) {
```
This if seems obsolete since this method is never called if the Erasertool is in use.

khalid-nasralla replied:
I removed the obsolete changes in the function.

khalid-nasralla commented:
I also refactored the tests that I wrote from java to kotlin, since the files where the tests are located, were refactored from java to kotlin on develop, causing merge conflicts. now they are resolved and I also fixed an old failing test that had the annotation "Ignored", now it passes again.

Lenkomotive reviewed (approved):
Good work! 👍🏼

SebastianGrief reviewed (approved):
LGTM 👍
